src/gurt/shm_lru.c

```diff
@@ -0,0 +1,535 @@
+/**
+ * (C) Copyright 2025 Hewlett Packard Enterprise Development LP
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <pthread.h>
+#include <errno.h>
+
+#include "shm_internal.h"
+#include <gurt/shm_utils.h>
+
+/* the address of shared memory region */
+extern struct d_shm_hdr *d_shm_head;
+
+/* Create a new LRU node */
+static int
+lru_create_node(shm_lru_cache_t *cache, shm_lru_cache_var_t *subcache, void *key, uint32_t key_size,
+		void *data, uint32_t data_size, shm_lru_node_t **new_node)
+{
+	shm_lru_node_t *node_list;
+	shm_lru_node_t *node;
+	char           *buf_key  = NULL;
+	char           *buf_data = NULL;
+	long int        idx_node;
+
+	*new_node = NULL;
+
+	/* make sure LRU is not full */
+	if (subcache->first_av < 0)
```

Comment:
I do not understand how an LRU can be full. by definition it should kick the oldest element out.
did i misunderstand this here?

Reply:
i see.. we probably should do some dynamic solution there.
meanwhile the put should document that on EBUSY one needs to  retry or just disregard putting the element as this is not fatal.

Reply:
Thank you! I will add the document as you suggested in another PR.
Regarding "some dynamic solution", could you please provide more information if there is any suggestion in your mind? Thank you!